test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs

```diff
@@ -96,6 +104,170 @@ public void CanChangeEngineFromScriptEditorDropDown()
+            var nodeModel = nodeView.ViewModel.NodeModel as PythonNodeBase;
+            Assert.NotNull(nodeModel);
+
+            //Get the Watch node so we can check the content later
+            var watchOUT = Model.CurrentWorkspace.NodeFromWorkspace<Watch>("714838b43a114f5f93cfe2f1d26092cf");
+
+            var scriptWindow = EditPythonCode(nodeView, View);
+            var codeEditor = FindCodeEditor(scriptWindow);
+
+            //Replace dictionary values so we can use the Revert and Run buttons
+            codeEditor.Text = codeEditor.Text.Replace("Autodesk", "Softdesk").Replace("1982", "1997");
+            codeEditor.Focus();
+
+            //This will get the buttons from the WPF xaml code
+            var saveButton = scriptWindow.FindName("SaveScriptChangesButton") as Button;
+            var runButton = scriptWindow.FindName("RunPythonScriptButton") as Button;
+            var revertButton = scriptWindow.FindName("RevertScriptChangesButton") as Button;
+
+            //Press the run button
+            runButton.RaiseEvent(new RoutedEventArgs(ButtonBase.ClickEvent));
+            DispatcherUtil.DoEvents();
+
+            //Check that the changes are reflected in the Watch
+            Assert.That(watchOUT.CachedValue,Is.EqualTo("Softdesk,1997"));
+
+            //Pressing the Revert button
+            revertButton.RaiseEvent(new RoutedEventArgs(ButtonBase.ClickEvent));
+            DispatcherUtil.DoEvents();
+            ViewModel.HomeSpace.Run();
+            
+            //After pressing the Revert button we need to have the previous result in Watch node
+            Assert.That(watchOUT.CachedValue, Is.EqualTo("Autodesk,1982"));
+
+            codeEditor.Text = codeEditor.Text.Replace("Softdesk","Autodesk").Replace("1997", "1982");
+            saveButton.RaiseEvent(new RoutedEventArgs(ButtonBase.ClickEvent));
+            ViewModel.HomeSpace.Run();
+
+            Assert.That(watchOUT.CachedValue, Is.EqualTo("Autodesk,1982"));
+        }
+
+        /// <summary>
+        /// This test method will validate that the OnMoreInfoEvent was executed correctly
+        /// </summary>
+        [Test]
+        public void OnMoreInfoEventTest()
+        {
+            Open(@"core\python\python_check_output.dyn");
+            ViewModel.HomeSpace.Run();
+
+            var nodeView = NodeViewWithGuid("3bcad14e-d086-4278-9e08-ed2759ef92f3");
+            var nodeModel = nodeView.ViewModel.NodeModel as PythonNodeBase;
+            Assert.NotNull(nodeModel);
+
+            //Get the Watch node so we can check the content later
+            var watchOUT = Model.CurrentWorkspace.NodeFromWorkspace<Watch>("714838b43a114f5f93cfe2f1d26092cf");
+
+            var scriptWindow = EditPythonCode(nodeView, View);
+            var codeEditor = FindCodeEditor(scriptWindow);
+            codeEditor.Text = codeEditor.Text.Replace("Autodesk", "Softdesk").Replace("1982", "1997");
```

Comment:
hmm what does this line do?

Reply:
@QilongTang 
This line is replacing the python dictionary values so later after re-running we will see the changes updated in the Watch node. I added an extra comment clarifying this info.